doc/extensions/htmx.rst

```diff
@@ -0,0 +1,71 @@
+htmx
+====
+
+Starting version 2.11, CKAN is shipped with `htmx <https://htmx.org/>`_.
+
+    *"htmx gives you access to AJAX, CSS Transitions, WebSockets and Server Sent
+    Events directly in HTML, using attributes, so you can build modern user
+    interfaces with the simplicity and power of hypertext."*
+    -- `htmx.org <https://htmx.org/>`_
+
+While not all CKAN templates have been updated to use htmx, you can use it
+in your own extensions to build modern user interfaces. htmx will be the core
+component in the implementation of the new CKAN UI, so you should expect more
+of it in future versions.
+
+---------
```

Comment:
I think these docs are good primer but we need to provide a bit more guidance.

An Overview section should give a general overview of how htmx works (hx- attributes in templates, background requests to retrieve updated HTML code that gets replaced in the page, etc)

Next, an Implementation section should go step by step explaining the different parts you need to implement, and we can use the Follow/Unfollow example for that but rather than including lots of code without much comment, explaining a bit more what is happening at each step. The `request.htmx` reference is less important and should be included last. For instance, when you added:

> The **Follow** button is a simple link that makes a POST request to the
``/dataset/follow/<dataset-id>`` and then replaces the section ``package-info`` with
all the HTML returned by the endpoint.

We can add:

"Note how we use the ``hx-post`` and ``hx-target`` attributes in the HTML tag of the link to configure htmx to perform these actions"

Just to point the user to the actual changes. I don't think we need to include the whole snippet in the docs, and on the view function we just need to highlight that we are returning a renderer template instead of a response.

As we use different approaches like a single view function handling htmx and regular requests we can add these too to the docs.

I'm happy to help writing these.

Reply:
@amercader check the new changes and let me know if you like them!